web-common/src/main/java/org/jboss/as/web/common/VirtualHttpServerMechanismFactoryMarkerUtility.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.web.common;
+
+import org.jboss.as.server.deployment.AttachmentKey;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * Utility class to mark a {@link DeploymentUnit} as requiring a virtual HttpServerAuthenticationMechanismFactory.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public class VirtualHttpServerMechanismFactoryMarkerUtility {
```

Comment:
I assume this helps free up the need to make changes to WildFly Core?

Reply:
Yes.